spec/unit/pops/types/iterable_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
+require 'puppet/pops'
+
+module Puppet::Pops::Types
+describe 'The enumeration support' do
+  it 'produces an iterable for Hash' do
+    expect(Iterable.on({:a=>1})).to respond_to(:step)
```

Comment:
Can this be made more precise? Or is the entire Iterable API based on object having a step method?